src/a2a/server/apps/jsonrpc/jsonrpc_app.py

```diff
@@ -154,22 +157,18 @@ class JSONRPCApplication(ABC):
     """
 
     # Method-to-model mapping for centralized routing
-    A2ARequestModel = (
-        SendMessageRequest
-        | SendStreamingMessageRequest
-        | GetTaskRequest
-        | CancelTaskRequest
-        | SetTaskPushNotificationConfigRequest
-        | GetTaskPushNotificationConfigRequest
-        | ListTaskPushNotificationConfigRequest
-        | DeleteTaskPushNotificationConfigRequest
-        | TaskResubscriptionRequest
-        | GetAuthenticatedExtendedCardRequest
-    )
-
-    METHOD_TO_MODEL: dict[str, type[A2ARequestModel]] = {
-        model.model_fields['method'].default: model
-        for model in A2ARequestModel.__args__
+    # Proto types don't have model_fields, so we define the mapping explicitly
+    METHOD_TO_MODEL: dict[str, type] = {
+        'message/send': SendMessageRequest,
+        'message/stream': SendStreamingMessageRequest,
+        'tasks/get': GetTaskRequest,
+        'tasks/cancel': CancelTaskRequest,
+        'tasks/pushNotificationConfig/set': SetTaskPushNotificationConfigRequest,
+        'tasks/pushNotificationConfig/get': GetTaskPushNotificationConfigRequest,
+        'tasks/pushNotificationConfig/list': ListTaskPushNotificationConfigRequest,
+        'tasks/pushNotificationConfig/delete': DeleteTaskPushNotificationConfigRequest,
+        'tasks/resubscribe': TaskResubscriptionRequest,
+        'agent/authenticatedExtendedCard': GetExtendedAgentCardRequest,
     }
```

Comment:
The method names have changed to align with the gRPC ones, i.e message/send is now "SendMessage". We need to fix that in the client and here.